x-pack/plugins/observability_solution/observability_onboarding/public/assets/auto_detect.sh

```diff
@@ -0,0 +1,177 @@
+#!/bin/bash
+
+log_file_list_string=""
+known_integrations_list_string=""
+selected_known_integrations_array=()
+selected_known_integrations_tsv_string=""
+unknown_log_file_path_list_string=""
+unknown_log_file_pattern_list_string=""
+selected_unknown_log_file_pattern_array=()
+selected_unknown_log_file_pattern_tsv_string=""
+custom_log_file_path_list_tsv_string=""
+
+read_log_file_list() {
+  local exclude_patterns=("^\/Users\/.+?\/Library\/Application Support")
+
+  local list=$(lsof -Fn | grep "log$" | awk '/^n/ {print substr($0, 2)}' | sort | uniq)
+  local filtered_list=""
+
+  # Filtering by the exclude patterns
+  while IFS= read -r line; do
+      if ! grep -qE "$(IFS="|"; echo "${exclude_patterns[*]}")" <<< "$line"; then
+          filtered_list+="$line\n"
+      fi
+  done <<< "$list"
+
+  log_file_list_string=$filtered_list
+}
+
+detect_known_integrations() {
+  while IFS= read -r log_file_path; do
+    local integration=""
+
+    if [[ $log_file_path =~ ^(/private)?/var/log/nginx ]]; then
+      integration="nginx"
+    fi
+    if [[ $log_file_path =~ ^(/private)?/var/log/(apache2|httpd) ]]; then
+      integration="apache"
+    fi
+    if [[ $log_file_path =~ ^(/private)?/var/lib/docker/containers ]]; then
+      integration="docker"
+    fi
+    if [[ $log_file_path =~ ^(/private)?/var/log/(syslog|auth|system|messages|secure) ]]; then
```

Comment:
Good points, thank you.

The `/private` thing should be ok, macOS stores actual files inside`/private` but has a symlink to `/var`, so `lsof` output gives paths with `/private` prefix but Elastic Agent will be able to collect logs directly from `/var`.

About the file pattern, we could expand to parse all open files, you're right, and then filter them down by `.log` extension for the custom logs.

Reply:
Ok, I've tried to expand the `lsof` command, but it produces to much noise besides open files (network ports, sockets, etc.) and there seems to be no way to limit it to actual files unfortunately.

Also, I realized that the current approach should work fine: the integration expects for example `[/var/log/messages*', '/var/log/syslog*', '/var/log/system*]` and as long as there is at least one `.log` file open in any of those locations we will detect the integration because we're matching only the leading part of the path, `^(/private)?/var/log/syslog` will match `/var/log/syslog*` pattern from the integration.

Reply:
Do we need to use lsof for known locations? 

I think we can simply check the presence of these folders/files. 

`lsof` would simply be used to detect any additional log files that haven't been already detected using the known file paths and then we can restrict the search to `*.log` files.